src/targz_sizes.c

```diff
@@ -120,8 +126,24 @@ int main(int argc, char** argv) {
         }
     }
 
+    unsigned int max_filename_blocks;
+    if ( LEVEL_IS_TEST( verbosity ) ) {
+        // when targz_sizes is in *test* mode, set max_filename_blocks
+        // very small (2).  This makes it easier to test the case where
+        // a long filename would overrun the buffer
+        max_filename_blocks=2;
+    } else {
+        // normally we set max_filename_blocks large
+        max_filename_blocks=200;
+    }
+    LOG_INFO(verbosity, "max_filename_blocks=%d", max_filename_blocks );
+
     unsigned char compressed[COMPRESSED_BUFFER];
     unsigned char decompressed[TAR_BLOCK_SIZE];
+    tarblock_t *filename_buff;
+    LOG_INFO(verbosity, "sizeof(tarblock_t)=%d", sizeof(tarblock_t) );
+    filename_buff = (tarblock_t*) calloc(
```

Comment:
In C99 you can make a local array with size given by another variable. Here you can just write `char filename_buff[max_filename_blocks * sizeof(tarblock_t)]`. Then you don't need to remember to `free()` it.